### PR TITLE
*: attach txn-info to complete operations

### PR DESCRIPTION
**Pull request by zyguan**
The new impl for #41

### CODE REVIEW

**Comment by zyguan** [comment]
[here](http://172.16.4.191/blue/rest/organizations/jenkins/pipelines/jepsen-test/runs/27834/nodes/53/steps/62/log/?start=0) is an example log.

**Comment by ekexium** [comment]
LGTM

**Comment by cfzjywxk** [comment]
@MyonKeminta PTAL

File: tidb/src/tidb/sql.clj [hunk 1]
```diff
@@ -246,36 +247,13 @@
 (defmacro with-txn
   "Executes body in a transaction, with a timeout, automatically retrying
   conflicts and handling common errors."
-  [op [c conn opts] & body]
+  [op opts & body]
```
Comment (cfzjywxk) [inline tidb/src/tidb/sql.clj:250]: What is the connection management mechanism here, not quite understand, will we meet the connection problem like before like doing rollback?

Comment (zyguan) [reply tidb/src/tidb/sql.clj:250]: No, I think. Previously a connection may be shared and accessed by two thread: the thread of worker (where we call `attach-txn-info` in `invoke!`, it's outside of `with-txn`) and the thread spawned by `timeout` (within `with-txn`), and now we just try to obtain txn info and attach it to `op` inside of the `timeout` thread.

**Comment by cfzjywxk** [comment]
LGTM

**Review by you06 — APPROVED** [review]
LGTM

**Pull request merged**